MonoGame.Framework/Utilities/ReflectionHelpers.Default.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Xna.Framework.Utilities
+{
+    internal static partial class ReflectionHelpers
+    {
+        /// <summary>
+        /// Generics handler for Marshal.SizeOf
+        /// </summary>
+        internal static class SizeOf<T>
+        {
+            static int _sizeOf;
+
+            static SizeOf()
+            {
+                _sizeOf = Marshal.SizeOf<T>();
```

Comment:
You know I was thinking that `Marshal.SizeOf<T>()` would internally do the same optimization we're doing here.  But actually it doesn't:

https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Runtime/InteropServices/Marshal.cs#L256

I don't see why they even made this change really... seems pointless.